pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -230,11 +212,25 @@ func (ctrl *VMSnapshotController) updateVMSnapshotContent(content *snapshotv1.Vi
 				continue
 			}
 
-			err = ctrl.freezeGuestFSIfNeeded(vmSnapshot)
+			source, err := ctrl.getSnapshotSource(vmSnapshot)
+			if err != nil {
+				return 0, err
+			}
+
+			frozen, err := source.Frozen()
 			if err != nil {
 				return 0, err
 			}
 
+			if !frozen {
```

Comment:
freeze should only be when its online vm snapshot and you didnt check here if the vm is online

Reply:
as-is, freeze will return nil if offline (no VMI) but is probably better to be explicit